### ISSUE TITLE
Menu "button" isn't keyboard accessible

### ISSUE THREAD

**Issue by scottaohara** [CityOfBoston/boston.gov#155]
## Details:

The primary menu button is not actually a button and does not accept the expected keyboard functionality (space bar or enter) to activate it, rendering the menu visually inaccessible to keyboard users.  

URL: [All of boston.gov](http://boston.gov)
## Steps to reproduce:
1. Use any browser and attempt to tab to the menu button
2. When focused (if browser will allow, Safari 9.1 won't) attempt to hit <kbd>Enter</kbd> or <kbd>Space</kbd>
3. Nothing happens...

(edit: tested using Safari, Chrome, FireFox on OSX. & with Safari + VoiceOver.  If I use Safari's VO+Space command, the menu will open, but only by using that specific key combination while VO is on.  The label is still incorrectly announced as a 'group' and it can't be accessed with normal tabbing within Safari.)
## There are two solutions I see to this issue:
1. Update the element to work appropriately with keyboard controls & to be announced correctly to ATs
2. Completely hide this element from ATs.  

**Solution 1:**
The menu label element currently has a `type="button"` being set to it, but this isn't a valid attribute to place on labels. To appropriately announce this as a button, it should have a `role="button"` applied to it, set via the [navigation JS file](https://github.com/CityOfBoston/boston.gov/blob/develop/docroot/sites/all/themes/custom/boston/src/js/main-nav.js), so that without JavaScript. There should also be a function added to allow allow the <kbd>Space</kbd>, and <kbd>Enter</kbd> keys to open the menu.  When either of these keys are pressed, keyboard focus should be moved to the parent element of the navigation, or should be set to the first focusable item within the nav.  Additionally, it'd be a nice-to-have to allow <kbd>Esc</kbd> to be pressed at anytime within the navigation to close the nav and return focus to the menu button.

Finally, it would make a lot more sense for the menu button to appear in the DOM before the navigation, since the current implementation requires a user to navigate through the visually hidden navigation before they can even get to the menu button to visually reveal it.  This is a pretty annoying usability issue for sighted keyboard users who may not typically need the assistance of a screen reader.  This then would result in a new issue, where if a user did not wish to open the menu, under the current implementation they would still need to navigate past all the items of the navigation, as they are only visually hidden but not truly hidden.  What could be done to rectify this would be to set a rule of `visibility: hidden;` to the contents of the navigation, and reset to `visibility: visible;` once the nav is appropriately opened.  These rules would only be set if JavaScript was enabled, as we wouldn't want the navigation to be completely inaccessible if JS were turned off or blocked.  

**Solution 2:**
Remove the `tabindex` and current aria-attributes from the label and use `aria-hidden="true"` to hide the element from ATs all together. Since the navigation is only visually hidden and is still completely accessible via normal keyboard controls, what could instead be done here is the navigation script could be updated to instead look to see if any of its children are ever the active element (have keyboard focus applied to them).  If so, then the navigation could have a class toggled that would visually bring it into view.  Once keyboard focus left the navigation, the class would then be removed and the navigation would hide again.

There would be no need to change functionality for sighted mouse users.

**For both solutions:**
The `tabindex` and `aria-expanded` attributes should not be hard-coded into the DOM, as without JavaScript the label can't be activated, and even if it could, the aria-expanded attribute would never be updated, and would only ever announce that the menu was closed.

I realize this is quite long, so I appreciate the time you have taken to get to this point :) Please let me know if there's anything I can clarify here.
## Browser Details

OSX: 
Safari 10.0.1, & Voice Over
FireFox 49.02, 
Chrome  54.0.2840.71 (64-bit)

**Comment by matthewcrist** [comment]
This is great @scottaohara. Thanks for taking the time to go into detail. We're in the process of looking at alternatives to the hamburger nav overall, but we'll work on implementing one of these solutions in the next couple of weeks.

**Comment by matthewcrist** [comment]
Opting for option 2 here. In the near future the menu will be open by default on large screens, but option 2 will still be a nice touch for smaller screens.

**Comment by matthewcrist** [comment]
This is live on Boston.gov.
